### PR TITLE
[DOC] Update loading action code example

### PR DESCRIPTION
**Pull request by nickeidler**
Added a guard against `undefined` controller in the `loading()` action documentation. Also updated syntax from using `set()` to directly setting values. See issue #19299 for context.

### CODE REVIEW

File: packages/@ember/-internals/routing/lib/system/route.ts [hunk 1]
```diff
@@ -500,12 +500,15 @@ class Route extends EmberObject implements IRoute {
...
       @action
       loading(transition, route) {
-        let controller = this.controllerFor('foo');
-        controller.set('currentlyLoading', true);
+        const controller = this.controllerFor('foo');
 
-        transition.finally(function() {
-          controller.set('currentlyLoading', false);
-        });
+        if (controller) {
```
Comment (acorncom) [inline packages/@ember/-internals/routing/lib/system/route.ts:505]: @nickeidler makes sense that controllers aren't always instantiated now that I stop to think about it, but it doesn't seem immediately apparent. Might be worth a code comment above this line explaining that briefly. Something like:

```suggestion
        // controller may not be instantiated when initially loading
        if (controller) {
```

Comment (nickeidler) [reply packages/@ember/-internals/routing/lib/system/route.ts:506]: Good point-- added a comment.

**Review by acorncom — APPROVED** [review]
LGTM. @locks any additional thoughts?

File: packages/@ember/-internals/routing/lib/system/route.ts [hunk 2]
```diff
@@ -500,12 +500,16 @@ class Route extends EmberObject implements IRoute {
     export default class ApplicationRoute extends Route {
       @action
       loading(transition, route) {
-        let controller = this.controllerFor('foo');
-        controller.set('currentlyLoading', true);
+        const controller = this.controllerFor('foo');
```
Comment (acorncom) [inline packages/@ember/-internals/routing/lib/system/route.ts:503]: @nickeidler I'd actually expect that we'll want this to go back to a `let`, Ember tends to default to a liberal let approach (https://madhatted.com/2016/1/25/let-it-be)

**Comment by locks** [comment]
Let's get it merged. Thanks!

**Pull request merged**